### PR TITLE
Add context support

### PR DESCRIPTION
**Pull request by paco0x**
Implement all the missing context support in `client/orm`

### CODE REVIEW

File: client/orm/cmd.go [hunk 1]
```diff
@@ -76,6 +77,7 @@ func RunCommand() {
 
 // sync database struct command interface.
 type commandSyncDb struct {
+	ctx       context.Context
```
Comment (flycash) [inline client/orm/cmd.go:80]: I think we don't need this `ctx` field. We could pass `context.Backgroud` if we need context.

From my understanding, we can do nothing with context when we run command fron temminal.

File: client/orm/orm_queryset.go [hunk 2]
```diff
@@ -64,22 +64,21 @@ func ColValue(opt operator, value interface{}) interface{} {
 
 // real query struct
 type querySet struct {
```
Comment (flycash) [inline client/orm/orm_queryset.go:66]: We should not using `context.Context` as field, it's not good practice.

And for this structure, the key point is to change `QuerySeter` interface. There are many methods in this interface but most of them don't need the `context` parameter.

But the "teminal method", which means that actually do query to DB, need to add parameter `ctx context.Context` to its signature.

Those methods are changed in this file now, you could check it.

But in order to keep compatiable, we should think about add new method, for example:
```go
Exist()
ExistWitCtx()
```

File: client/orm/types.go [hunk 3]
```diff
@@ -235,6 +235,8 @@ type Inserter interface {
 
 // QuerySeter query seter
 type QuerySeter interface {
+	// add query context for querySeter
+	WithContext(context.Context) QuerySeter
```
Comment (flycash) [inline client/orm/types.go:239]: As mentioned before, we don't need this method.

**Review by flycash — CHANGES_REQUESTED** [review]
I think using context.Context as strucure's field is not a good practice.

File: client/orm/cmd.go (hunk 1 above)
Comment (paco0x) [reply client/orm/cmd.go:80]: this can be removable

File: client/orm/orm_queryset.go (hunk 2 above)
Comment (paco0x) [reply client/orm/orm_queryset.go:66]: In some aspects, I agree with you. But idk what's the plan for your team on this orm module. Your comment seems inconsistent with the existing code. 

You can see there's already `QueryM2MWithCtx` and `QueryTableWithCtx` method in the codebase.  If we remove `ctx` field on `querySet` struct. These two methods and related code also need to be removed. I reserved the `ctx` just for compatibility with the existing code.

I can do a revise as you suggested. But just wanna make sure what the idea of the guy who wrote and merged `QueryM2MWithCtx` and `QueryTableWithCtx`.

File: client/orm/types.go (hunk 3 above)
Comment (paco0x) [reply client/orm/types.go:239]: Same as the above comment.

File: client/orm/orm_queryset.go (hunk 2 above)
Comment (flycash) [reply client/orm/orm_queryset.go:66]: let me check that.

Comment (flycash) [reply client/orm/orm_queryset.go:66]: I believe this is my mistake. 
I think one way is that keep both two methods and your solution is acceptable. But as I said that, using `ctx` as structure's field is not a good practice.
Another way is that mark both `QueryM2MWithCtx` and `QueryTableWithCtx` as deprecated. In this case, we want you remove the ctx field.

What do you think?

Comment (flycash) [reply client/orm/orm_queryset.go:66]: If we think querySeter is "request-scope", using ctx as structure field looks not bad.

Comment (paco0x) [reply client/orm/orm_queryset.go:66]: To me, I'd prefer the suggestion on the first comment you make. That is to make `context.Context` parameter on the `Insert/Delete/Update` method on `QuerySetter`. I think it's good for any use case cause we don't know how users gonna use it in "request-scope" or something other. 

For the compatibility side. I saw these two `QueryM2MWithCtx ` and `QueryTableWithCtx ` methods are already in the master branch and some tagged versions. I'll push another commit to remove all the `ctx` filed and mark that two methods as deprecated.

**Comment by paco0x** [comment]
done. the sonarcloud code analysis seems to keep failing

File: client/orm/orm_object.go [hunk 4]
```diff
@@ -15,12 +15,14 @@
...
 
 import (
+	"context"
 	"fmt"
 	"reflect"
 )
 
 // an insert queryer struct
 type insertSet struct {
+	ctx    context.Context
```
Comment (flycash) [inline client/orm/orm_object.go:25]: it's similar to QuerySter, and I think we could remove this field and trigger it like other interface.

**Review by flycash — CHANGES_REQUESTED** [review]
Good, and the insertSet and sonar need to be fixed.

**Comment by flycash** [comment]
Oh, you could ignore Sonar problem. It's a little big hard to fix them.

File: client/orm/orm_object.go (hunk 4 above)
Comment (paco0x) [reply client/orm/orm_object.go:25]: done

**Review by flycash — APPROVED** [review]

**Pull request merged**